bundles/binding/org.openhab.binding.mailcontrol/src/main/java/org/openhab/binding/mailcontrol/internal/MailControlGenericBindingProvider.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright (c) 2010-2013, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.mailcontrol.internal;
+
+import org.openhab.binding.mailcontrol.MailControlBindingProvider;
+import org.openhab.core.binding.BindingConfig;
+import org.openhab.core.items.Item;
+import org.openhab.core.items.ItemRegistry;
+import org.openhab.model.item.binding.AbstractGenericBindingProvider;
+import org.openhab.model.item.binding.BindingConfigParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * This class is responsible for parsing the binding configuration.
+ * 
+ * @author Andrey.Pereverzin
+ * @since 1.7.0
+ */
+public class MailControlGenericBindingProvider extends AbstractGenericBindingProvider implements MailControlBindingProvider {
+    // Injected by the OSGi Container through the setItemRegistry and
+    // unsetItemRegistry methods.
+    private ItemRegistry itemRegistry;
+
+    private static final Logger logger = LoggerFactory.getLogger(MailControlGenericBindingProvider.class);
+    
+	/**
+	 * {@inheritDoc}
+	 */
+	public String getBindingType() {
+		return "mailcontrol";
+	}
+
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	public void validateItemType(Item item, String bindingConfig) throws BindingConfigParseException {
+		//if (!(item instanceof SwitchItem || item instanceof DimmerItem)) {
+		//	throw new BindingConfigParseException("item '" + item.getName()
+		//			+ "' is of type '" + item.getClass().getSimpleName()
+		//			+ "', only Switch- and DimmerItems are allowed - please check your *.items configuration");
+		//}
```

Comment:
please implement this check on ItemType base. If all types are allowed please document it